subprojects/tooling-api/src/integTest/groovy/org/gradle/integtests/tooling/fixture/CrossVersionToolingApiSpecificationRetryRuleTest.groovy

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.tooling.fixture
+
+import org.gradle.util.Requires
+
+class CrossVersionToolingApiSpecificationRetryRuleTest extends ToolingApiSpecification {
+
+    def setup() {
+        //these meta tests mess with the daemon log: do not interfere with other tests when running in parallel
+        toolingApi.requireIsolatedDaemons()
+    }
+
+    def iteration = 0
+
+    @Requires(adhoc = {ToolingApiSpecification.runsOnWindowsAndJava7()})
+    def "retries when expected exception occurs"() {
```

Comment:
Nice test coverage.  Should we also have a test to check that we don't retry when not running on windows/Java7?